tests/integration/moderation.test.ts

```diff
@@ -1,184 +1,187 @@
+  createNewReport,
+  createNewTestimony,
+  createReqObj,
+  signInTestAdmin,
+  signInUser
+} from "tests/integration/common"
+import { auth } from "../../components/firebase"
+import e from "express"
+import handler from "pages/api/users/[uid]/testimony/[tid]"
+import { NextApiRequest, NextApiResponse } from "next"
+import { mapleClient } from "components/api/maple-client"
+// import supertest from "supertest"
 // afterAll(terminateFirebase)
 
-const authtoken = process.env.AUTH_TOKEN
+// const authtoken = process.env.AUTH_TOKEN
 
-let testimonyRef: DocumentReference<DocumentData>
-let reportRef: DocumentReference<DocumentData>
 let adminUid: string
-let userId: string
+let uid: string
 
-beforeAll(async () => {
-  const adminUser = await testAuth.createUser(fakeUser())
-  const role = "admin"
-  await expect(
-    testAuth.setCustomUserClaims(adminUser.uid, { role: role })
-  ).resolves.toBeUndefined()
-
-  const updatedUser = await testAuth.getUser(adminUser.uid)
-  adminUid = updatedUser.uid
-
-  const { user, testimony, report } = createFakeTestimonyReport()
-
-  userId = user.uid
-
-  expect(userId).toEqual(testimony.authorUid)
+// const MockGoogleAuth = jest.mock("../../node_modules/google-auth-library")
 
-  const userRef = testDb.doc(`/users/${userId}`)
-  userRef.set({ id: userId })
+beforeAll(async () => {
+  const user = await signInTestAdmin()
+  uid = user.uid
+  adminUid = user.uid
 
-  testimonyRef = testDb.doc(
-    `/users/${userId}/publishedTestimony/${testimony.id}`
-  )
+  await db.doc(`/profiles/${uid}`).update({ role: "admin" })
+  expect((await db.doc(`/profiles/${uid}`).get()).data()?.role).toEqual("admin")
 
-  reportRef = testDb.doc(`reports/${report.id}`)
+  expect(auth.currentUser).toBeDefined()
+})
 
-  await testDb.doc(testimonyRef.path).set(testimony)
-  await testDb.doc(reportRef.path).set(report)
+describe("this", () => {
+  it("testing the set up", async () => {})
 })
 
-describe("reporting testimony flow", () => {
-  it("testimony and report exist", async () => {
-    const testimonyResult = await testimonyRef.get()
-    expect(testimonyResult.exists).toBeTruthy()
+describe("examples of helper functions", () => {
+  it("shows it's logged in as admin", async () => {
+    try {
+      expect((await adminAuth.getUser(adminUid)).customClaims).toEqual({
+        role: "admin"
+      })
+    } catch (e) {
+      console.warn(e)
+    }
+
+    expect(await adminAuth.getUser(adminUid)).toBeDefined()
+    expect((await adminAuth.getUser(adminUid)!).uid).toEqual(adminUid)
+  })
 
-    const reportResult = await testimonyRef.get()
-    expect(reportResult.exists).toBeTruthy()
+  it("creates testimony", async () => {
+    const { tid, getThisTestimony, whereIsThisTestimony, removeThisTestimony } =
+      await createNewTestimony(uid, "H1002")
+    const test1 = await getThisTestimony()
+    expect(test1).toBeDefined()
+    expect(test1!.id).toEqual(tid)
+
+    // clean up
+    await removeThisTestimony()
```

Comment:
I like how you factored out common actions into its own function, however I fear within tests it's not clear what's actually being tested.

Is there a particular "maple-specific" function not created just to help out tests that this is testing? If not, we might exclude these tests because we don't need to test firebase functions.

Reply:
However looking at other tests those functions seem pretty useful to test the firebase rules, etc.